packages/commonwealth/client/scripts/views/components/sidebar/external_links_module.tsx

```diff
@@ -34,60 +45,158 @@ export const ExternalLinksModule = () => {
+            className="tiktok-link"
+            onClick={() => setSelectedLink(link)}
+          />
+          {selectedLink === link && (
+            <CWModal
+              size="small"
+              content={
+                <SocialAccountLinkModal
+                  onModalClose={handleModalClose}
+                  formattedLink={formatLink(link)}
+                />
+              }
+              onClose={handleModalClose}
+              open={true}
+            />
+          )}
+        </React.Fragment>
       ))}
       {twitters.map((link) => (
-        <CWIcon
-          key={link}
-          iconName="twitterX"
-          className="twitter-link"
-          onClick={() => window.open(link)}
-        />
+        <React.Fragment key={link}>
+          <CWIcon
+            iconName="twitterX"
+            className="twitter-link"
+            onClick={() => setSelectedLink(link)}
+          />
+          {selectedLink === link && (
+            <CWModal
+              size="small"
+              content={
+                <SocialAccountLinkModal
+                  onModalClose={handleModalClose}
+                  formattedLink={formatLink(link)}
+                />
+              }
+              onClose={handleModalClose}
+              open={true}
+            />
+          )}
+        </React.Fragment>
       ))}
       {githubs.map((link) => (
-        <CWIcon
-          key={link}
-          iconName="github"
-          className="github-link"
-          onClick={() => window.open(link)}
-        />
+        <React.Fragment key={link}>
+          <CWIcon
+            iconName="github"
+            className="github-link"
+            onClick={() => setSelectedLink(link)}
+          />
+          {selectedLink === link && (
+            <CWModal
+              size="small"
+              content={
+                <SocialAccountLinkModal
+                  onModalClose={handleModalClose}
+                  formattedLink={formatLink(link)}
+                />
+              }
+              onClose={handleModalClose}
+              open={true}
+            />
+          )}
+        </React.Fragment>
       ))}
       {[...remainingLinks, ...slacks].map((link) => (
-        <CWIcon
-          key={link}
-          iconName="website"
-          className="website-link"
-          onClick={() => window.open(link)}
-        />
+        <React.Fragment key={link}>
+          <CWIcon
+            iconName="website"
+            className="website-link"
+            onClick={() => setSelectedLink(link)}
+          />
+          {selectedLink === link && (
+            <CWModal
+              size="small"
+              content={
+                <SocialAccountLinkModal
+                  onModalClose={handleModalClose}
+                  formattedLink={formatLink(link)}
+                />
+              }
+              onClose={handleModalClose}
+              open={true}
+            />
+          )}
+        </React.Fragment>
```

Comment:
please remove all those CWModal, you just need one instance probably here below in line 200. Just adjust open=true to be dynamic